libyul/optimiser/StructuralSimplifier.cpp

```diff
@@ -45,6 +128,20 @@ void StructuralSimplifier::operator()(Block& _block)
 	popScope();
 }
 
+OptionalStatements StructuralSimplifier::reduceNoCaseSwitch(Switch& _switchStmt) const
+{
+	yulAssert(_switchStmt.cases.empty(), "Expected no case!");
+
+	auto loc = locationOf(*_switchStmt.expression);
+
+	OptionalStatements s = vector<Statement>{};
+
+	if (!hasLiteralValue(*_switchStmt.expression))
```

Comment:
I think you could instead use ``MovableChecker`` here to catch a larger class of expressions... However: I guess we should have other components that remove ``pop(<movableExpr>)`` anyways (I hope at least ;-)), so we may even want to keep all these ``pop``s here and let them be removed elsewhere.

Reply:
Things should be maximally modular after all and I'm not sure whether removing ``pop(<movable>)`` or ``pop(<literal>)`` is in fact a "structural simplification" :-).

Reply:
Yep, `pop(<movable>)` is removed in the Unused Pruner.